src/hooks/__tests__/create.test.js

```diff
@@ -0,0 +1,79 @@
+describe('create hook test', () => {
+  beforeEach(() => {
+    jest.mock('/engine/create-auth-hook', () => ({
+      createAuthHook: jest.fn(),
+      onAuthorized: jest.fn()
+    }));
+  });
+
+  afterEach(() => {
+    jest.resetModules();
+  });
+
+  test('should call create hook proper options', async () => {
+    const { create } = require('../create');
+    const { createAuthHook } = require('../../engine/create-auth-hook');
+    const options = {
+      authenticate: jest.fn(),
+      ruleServiceName: 'authorization-rules',
+      getPrincipalId: jest.fn(),
+      getArguments: jest.fn(),
+      isAuthorized: jest.fn(),
+      onAuthorized: await jest.fn(),
+      attributeLookup: {
+        principalId: jest.fn()
+      }
+    };
+
+    create(options);
+
+    expect(createAuthHook).toHaveBeenCalledWith({
+      authenticate: expect.any(Function),
```

Comment:
We should test that the function that was passed as `authenticate` is the same one was put in the options object above; same with all the other attributes here.